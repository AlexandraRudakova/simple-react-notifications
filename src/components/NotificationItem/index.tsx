import React from "react";

const NotificationItem = (props: any) => <div>{props.text}</div>;

export default NotificationItem;
