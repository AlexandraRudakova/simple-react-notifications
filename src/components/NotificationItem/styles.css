@keyframes slideToRight {
  20% {
    opacity: 1;
    transform: translate3d(-20px, 0, 0);
  }
  to {
    opacity: 0;
    transform: translate3d(2000px, 0, 0);
  }
}

@keyframes slideToLeft {
  20% {
    opacity: 1;
    transform: translate3d(20px, 0, 0);
  }
  to {
    opacity: 0;
    transform: translate3d(-2000px, 0, 0);
  }
}

.notification-item {
  background: white;
  color: gray;
  position: relative;
  min-height: 64px;
  box-sizing: border-box;
  margin-bottom: 16px;
  padding: 8px;
  border-radius: 1px;
  box-shadow: 0 1px 10px 0 rgba(0, 0, 0, 0.1), 0 2px 15px 0 rgba(0, 0, 0, 0.05);
  display: flex;
  align-items: center;
  animation-timing-function: ease-in;
  animation-iteration-count: 1;
  animation-direction: normal;
  animation-fill-mode: forwards;
}

.top-left .notification-item,
.bottom-left .notification-item {
  animation-name: slideToLeft;
}

.top-right .notification-item,
.bottom-right .notification-item {
  animation-name: slideToRight;
}

.notification-item.info {
  background: #3498db;
  color: white;
}

.notification-item.alert {
  background: #e74c3c;
  color: white;
}

.notification-item.warning {
  background: #f1c40f;
  color: white;
}

.notification-item label {
  display: inline-block;
  width: 20px;
  text-align: center;
  font-size: 20px;
}

.notification-item span {
  margin: 0 20px 0 10px;
}

.notification-item button {
  cursor: pointer;
  color: white;
  background: transparent;
  border: 0;
  opacity: 0.7;
  position: absolute;
  top: 8px;
  right: 8px;
}
