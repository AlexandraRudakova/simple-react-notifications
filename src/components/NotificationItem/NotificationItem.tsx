import React from "react";
import "./styles.css";

type Props = {
  message: string;
  type?: string;
  animationDelay?: string;
  animationDuration?: string;
  onClose?: () => void;
};

const icons = {
  warning: "⚠",
  info: "ℹ",
  alert: "⚡"
};

const NotificationItem = ({
  message,
  type = "info",
  onClose = () => {},
  animationDelay = "0ms",
  animationDuration = "0ms"
}: Props) => (
  <div
    className={`notification-item ${type}`}
    style={{
      animationDelay,
      animationDuration
    }}
  >
    <label>{icons[type]}</label>
    <span role={type}>{message}</span>
    <button type="button" aria-label="close" onClick={onClose}>
      ✖
    </button>
  </div>
);

export default NotificationItem;
