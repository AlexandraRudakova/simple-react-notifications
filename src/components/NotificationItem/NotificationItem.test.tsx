import React from "react";
import { shallow } from "enzyme";
import NotificationItem from "./NotificationItem";

describe("<NotificationItem />", () => {
  it("should render correctly with all props", () => {
    const wrapper = shallow(
      <NotificationItem
        message="Hello"
        type="warning"
        animationDelay="5000ms"
        animationDuration="300ms"
        onClose={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("label").length).toEqual(1);
  });

  it("should render with all default props", () => {
    const wrapper = shallow(<NotificationItem message="Hello" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("label").length).toEqual(1);
    expect(wrapper.contains(<label>â„¹</label>)).toEqual(true);
  });
});
