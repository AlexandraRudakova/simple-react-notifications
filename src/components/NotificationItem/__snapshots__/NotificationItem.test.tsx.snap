// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationItem /> should render correctly with all props 1`] = `
<div
  className="notification-item warning"
  style={
    Object {
      "animationDelay": "5000ms",
      "animationDuration": "300ms",
    }
  }
>
  <label>
    ⚠
  </label>
  <span
    role="warning"
  >
    Hello
  </span>
  <button
    aria-label="close"
    onClick={[Function]}
    type="button"
  >
    ✖
  </button>
</div>
`;

exports[`<NotificationItem /> should render with all default props 1`] = `
<div
  className="notification-item info"
  style={
    Object {
      "animationDelay": "0ms",
      "animationDuration": "0ms",
    }
  }
>
  <label>
    ℹ
  </label>
  <span
    role="info"
  >
    Hello
  </span>
  <button
    aria-label="close"
    onClick={[Function]}
    type="button"
  >
    ✖
  </button>
</div>
`;
